src/lua/uri.lua

```diff
@@ -169,8 +171,52 @@ local function format(uri, write_password)
     return str
 end
 
+local function encode_kv(key, values, res)
+    local val = values
+    if type(val) ~= "table" then
+        val = { val }
+    end
+
+    -- { a = {} } --> "a"
+    if next(val) == nil then
+        table.insert(res, key)
+    end
+
+    -- { a = { "b" } } --> "a=c"
+    for _, v in pairs(val) do
+        local kv = ("%s=%s"):format(key, v)
+        table.insert(res, kv)
+    end
+end
+
+-- Encode map to a string.
+local function params(opts)
+    if not opts then
+        return ""
+    end
```

Comment:
I don't think this is a big deal, but
`params(false) == ""` while
`params(true)` will rise an error.

Reply:
```suggestion
    if opts == nil then
        return ""
    end
```